internal/util/observability/funccall.go

```diff
@@ -56,11 +58,16 @@ func FuncCall(ctx context.Context) func() {
 		fc.region = trace.StartRegion(ctx, funcName)
 	}
 
-	return fc.leave
+	stopCtx, stopCancel := context.WithCancel(context.Background())
+	go fc.run(stopCtx)
```

Comment:
FuncCall can't create new goroutines because `Region.End` method must be called from the same goroutine where the region was started